### DISCUSSION TITLE
How to remove all of default admin menus and add cutomizing admin menus,please?

### DISCUSSION THREAD

**Discussion by effapp** [OrchardCMS/OrchardCore#15498] [General]
How to remove all of default admin menus and add cutomizing admin menus,please?

**Comment by rwxzig** [comment]
You can edit the menus at Configuration / Admin Menus
![image](https://github.com/OrchardCMS/OrchardCore/assets/42944941/9dc6ec7b-9f90-41ec-bf12-249df9e44715)

**Reply by effapp** [reply]
Hi cptzig,
      "Admin Menus" can add menus,but can't remove initial admin menus.

**Comment by Skrypt** [comment]
You can also have a blank setup recipe that will not add any menus in there. If that's not enough you will need to work with a custom NavigationProvider (code).

**Reply by effapp** [reply]
Hi Skrype,
     The blank setup recipe will generate many initial admin menu items,too.

**Reply by Skrypt** [reply]
The modules will add admin menu items that are necessary for them. You can't remove them all. One way to remove them is to create a user role that has pretty much no permission and it will remove these menu items that are only necessary for admins. If that's the goal you are trying to reach. I would not mutate these necessary admin menu items but restrict them per user role.

**Reply by effapp** [reply]
I want to hide admin menu item "Content->ContentItems",and add customizing admin menu items which like "Content->Page", "Content->Project","Content->{{Special ContentType}}",etc...
I can add  customizing admin menu items through "Configuration->Admin Menus", but
How can I achieve hiding admin menu item "Content->ContentItems",please?

**Reply by Skrypt** [reply]
You will need to remove the 
services.AddScoped<INavigationProvider, AdminMenu>();
that is registered in the startup of OrchardCore.Contents.
I think you can acheive this by using a custom module with a startup that will do a services.Remove<> in it own startup.cs

I don't have all the details about this services.Remove though but you will need to remove the proper one that is coming from OrchardCore.Contents assembly.

**Comment by ns8482e** [comment]
Admin Menus are create using `INavigationProvider`. There is no built-in ways support completely change the admin menus. However there are different approach that you can take to add/modify 

- Hide the default menu shape altogether and you implement your own menu system - OR
- Remove all  `services.AddScoped<INavigationProvider, AdminMenu>();`  registration as @Skrypt  suggested. - OR
- Add/Register your `INavigationProvider` modifies the existing menus and add/remove/creates new menus as per your need
